packages/node-sdk/src/client.ts

```diff
@@ -0,0 +1,607 @@
+        `get request to "${path}" failed with error`,
+        error,
+      );
+      return undefined;
+    }
+  }
+
+  /**
+   * Sends a feature flag event to the Bucket API.
+   *
+   * @param event - The event to send.
+   *
+   * @returns A boolean indicating if the request was successful.
+   * @throws An error if the event is invalid.
+   **/
+  private async sendFeatureFlagEvent(event: FeatureFlagEvent) {
+    ok(typeof event === "object", "event must be an object");
+    ok(
+      typeof event.action === "string" &&
+        (event.action === "evaluate" || event.action === "check"),
+      "event must have an action",
+    );
+    ok(
+      typeof event.flagKey === "string" && event.flagKey.length > 0,
+      "event must have a flag key",
+    );
+    ok(
+      typeof event.flagVersion === "number" || event.flagVersion === undefined,
+      "event must have a flag version",
+    );
+    ok(
+      typeof event.evalResult === "boolean",
+      "event must have an evaluation result",
+    );
+    ok(
+      event.evalContext === undefined || typeof event.evalContext === "object",
+      "event context must be an object",
+    );
+    ok(
+      event.evalRuleResults === undefined ||
+        Array.isArray(event.evalRuleResults),
+      "event rule results must be an array",
+    );
+    ok(
+      event.evalMissingFields === undefined ||
+        Array.isArray(event.evalMissingFields),
+      "event missing fields must be an array",
+    );
+
+    return await this.post("flags/events", {
+      action: event.action,
+      flagKey: event.flagKey,
+      flagVersion: event.flagVersion,
+      evalContext: event.evalContext,
+      evalResult: event.evalResult,
+      evalRuleResults: event.evalRuleResults,
+      evalMissingFields: event.evalMissingFields,
+    });
+  }
+
+  private getFeatureFlagDefinitionCache() {
+    if (!this._shared.featureFlagDefinitionCache) {
+      this._shared.featureFlagDefinitionCache = cache<FlagDefinitions>(
+        this._shared.refetchInterval,
+        this._shared.staleWarningInterval,
+        this._shared.logger,
+        async () => {
+          const res = await this.get<FlagDefinitions>("flags");
+
+          if (!isObject(res) || !Array.isArray(res?.flags)) {
+            return undefined;
+          }
+
+          return res;
+        },
+      );
+    }
+
+    return this._shared.featureFlagDefinitionCache;
+  }
+
+  /**
+   * Sets the user that is used for feature flag evaluation.
+   *
+   * @param userId - The user ID to set.
+   * @param opts.attrs - The attributes of the user (optional).
+   *
+   * @returns A new client with the user set.
+   * @throws An error if the user ID is not a string or the options are invalid.
+   * @remarks
+   * If the user ID is the same as the current company, the attributes will be merged, and
+   * the new attributes will take precedence.
+   **/
+  public withUser(
+    userId: string,
+    opts?: { attributes?: Attributes },
```

Comment:
do you already have something in mind for the other options? If not, let's make this simpler and just let is take the attributes?: Attributes as a second parameter without the opts. If we really want to add more options later, that's still possible.